### ISSUE TITLE
[Bug] Crashlytics not properly starting on Unity 2020.3.0f1 when using 7.1.0

### ISSUE THREAD

**Issue by demianardus** [firebase/quickstart-unity#1011]
### [REQUIRED] Please fill in the following fields:

  * Unity editor version: 2020.3.0f1
  * Firebase Unity SDK version: 7.1.0
  * Source you installed the SDK: .unitypackage (dotnet4)
  * Problematic Firebase Component: Crashlytics
  * Other Firebase Components in use: Auth, Firestore
  * Additional SDKs you are using: GooglePlayGames [0.10.12]
  * Platform you are using the Unity editor on: Windows
  * Platform you are targeting: iOS and Android
  * Scripting Runtime: Mono

### [REQUIRED] Please describe the issue here:

I've been working on a game for a couple months, and I've been using Firebase Auth and Firestore without any issues. Yesterday, I tried to setup Crashlytics, so I followed every step on https://firebase.google.com/docs/crashlytics/get-started?platform=unity&authuser=0#add-sdk 

I imported the Crashlytics unity package from the same SDK download I imported the Auth and Firestore ones.

Now when I hit play on Unity, crashlytics does not seem to properly initialize, and if I further into the game make it crash, I do not see anything being reported.

I'm initializing firebase as I was already doing so in order to use Auth and Firestore.

#### Steps to reproduce:

Have you been able to reproduce this issue with just the Firebase Unity quickstarts (this GitHub project)? Did not try
What's the issue repro rate? -

What happened? How can we make the problem occur? 
I get the following debug log from firebase.

```
DEBUG: Deleting Auth 000001EAC2AB6DD0 for App 000001EA2E7DDF50
DEBUG: Deleting app __FIRAPP_DEFAULT (0x2e7ddf50)
DEBUG: Disabling all app initializers
DEBUG: Disable analytics
DEBUG: Disable auth
DEBUG: Disable database
DEBUG: Disable dynamic_links
DEBUG: Disable functions
DEBUG: Disable instance_id
DEBUG: Disable messaging
DEBUG: Disable remote_config
DEBUG: Disable storage
Native extension for WindowsStandalone target not found
Native extension for iOS target not found
Native extension for Android target not found
Refreshing native plugins compatible for Editor in 10.19 ms, found 5 plugins.
Preloading 0 native plugins for Editor in 0.00 ms.
UnityIAP: [InitializeOnLoad] Facebook Check
UnityIAP: Runtime [2020.3.0f1]
UnityIAP: Current Build is Standalone:StandaloneWindows
UnityIAP: stub FacebookStore.dll disabled
Invoked RoslynAnalysisRunner static constructor.
RoslynAnalysisRunner will not be running.
RoslynAnalysisRunner has terminated.
[MODES] ModeService[none].Initialize
[MODES] ModeService[none].LoadModes
[MODES] Loading mode Default (0) for mode-current-id-X
Mono: successfully reloaded assembly
- Completed reload, in  1.566 seconds
Platform modules already initialized, skipping
Initializing UnityPurchasing via Codeless IAP

UnityIAP Version: 2.2.7

Loaded scene 'Temp/__Backupscenes/0.backup'

DEBUG: Disabling all app initializers

DEBUG: Disable analytics

DEBUG: Disable auth

DEBUG: Disable database

DEBUG: Disable dynamic_links

DEBUG: Disable functions

DEBUG: Disable instance_id

DEBUG: Disable messaging

DEBUG: Disable remote_config

DEBUG: Disable storage

Enable module 'analytics' for 'Firebase.Analytics.FirebaseAnalytics, Firebase.Analytics'

DEBUG: analytics app initializer Enabling

Enable module 'auth' for 'Firebase.Auth.FirebaseAuth, Firebase.Auth'

DEBUG: auth app initializer Enabling

Enable module 'crashlytics' for 'Firebase.Crashlytics.Crashlytics, Firebase.Crashlytics'
UnityEngine.StackTraceUtility:ExtractStackTrace ()
UnityEngine.DebugLogHandler:LogFormat (UnityEngine.LogType,UnityEngine.Object,string,object[])
UnityEngine.Logger:Log (UnityEngine.LogType,object)
UnityEngine.Debug:Log (object)
Firebase.Platform.FirebaseLogger:LogMessage (Firebase.Platform.PlatformLogLevel,string) (at Z:/tmp/tmp.pKCgWTopY4/firebase/app/client/unity/src/Unity/FirebaseLogger.cs:89)
Firebase.LogUtil:LogMessage (Firebase.LogLevel,string) (at Z:/tmp/tmp.teccva8Fjv/firebase/app/client/unity/proxy/LogUtil.cs:68)
Firebase.FirebaseApp:InitializeAppUtilCallbacks () (at Z:/tmp/tmp.teccva8Fjv/firebase/app/client/unity/proxy/FirebaseApp.cs:433)
Firebase.FirebaseApp:CheckDependenciesAsync () (at Z:/tmp/tmp.teccva8Fjv/firebase/app/client/unity/proxy/FirebaseApp.cs:728)
Firebase.FirebaseApp:CheckAndFixDependenciesAsync () (at Z:/tmp/tmp.teccva8Fjv/firebase/app/client/unity/proxy/FirebaseApp.cs:784)
FirebaseInitializer:Start () (at Assets/Scripts/Firebase/FirebaseInitializer.cs:25)
(Filename: Assets/Scripts/Firebase/FirebaseInitializer.cs Line: 25)

DEBUG: App initializer crashlytics not found, failed to enable.
App initializer crashlytics not found, failed to enable.
UnityEngine.StackTraceUtility:ExtractStackTrace ()
UnityEngine.DebugLogHandler:LogFormat (UnityEngine.LogType,UnityEngine.Object,string,object[])
UnityEngine.Logger:Log (UnityEngine.LogType,object)
UnityEngine.Debug:Log (object)
Firebase.Platform.FirebaseLogger:LogMessage (Firebase.Platform.PlatformLogLevel,string) (at Z:/tmp/tmp.pKCgWTopY4/firebase/app/client/unity/src/Unity/FirebaseLogger.cs:89)
Firebase.LogUtil:LogMessage (Firebase.LogLevel,string) (at Z:/tmp/tmp.teccva8Fjv/firebase/app/client/unity/proxy/LogUtil.cs:68)
Firebase.LogUtil:LogMessageFromCallback (Firebase.LogLevel,string) (at Z:/tmp/tmp.teccva8Fjv/firebase/app/client/unity/proxy/LogUtil.cs:76)
Firebase.AppUtil:SetEnabledAppCallbackByName (string,bool) (at Z:/tmp/tmp.teccva8Fjv/firebase/app/client/unity/proxy/AppUtil.cs:53)
Firebase.FirebaseApp:InitializeAppUtilCallbacks () (at Z:/tmp/tmp.teccva8Fjv/firebase/app/client/unity/proxy/FirebaseApp.cs:439)
Firebase.FirebaseApp:CheckDependenciesAsync () (at Z:/tmp/tmp.teccva8Fjv/firebase/app/client/unity/proxy/FirebaseApp.cs:728)
Firebase.FirebaseApp:CheckAndFixDependenciesAsync () (at Z:/tmp/tmp.teccva8Fjv/firebase/app/client/unity/proxy/FirebaseApp.cs:784)
FirebaseInitializer:Start () (at Assets/Scripts/Firebase/FirebaseInitializer.cs:25)
(Filename: Assets/Scripts/Firebase/FirebaseInitializer.cs Line: 25)

DEBUG: database app initializer Disabling

DEBUG: dynamic_links app initializer Disabling

DEBUG: functions app initializer Disabling

DEBUG: App initializer installations not found, failed to enable.

DEBUG: instance_id app initializer Disabling

DEBUG: App initializer invites not found, failed to enable.

DEBUG: messaging app initializer Disabling

DEBUG: App initializer performance not found, failed to enable.

DEBUG: remote_config app initializer Disabling

DEBUG: storage app initializer Disabling

DEBUG: App initializer test_lab not found, failed to enable.

DEBUG: Creating Firebase App __FIRAPP_DEFAULT for Firebase C++ 7.1.0
Creating Firebase App __FIRAPP_DEFAULT for Firebase C++ 7.1.0
(Filename: Assets/Scripts/Firebase/FirebaseInitializer.cs Line: 29)

DEBUG: Added app name=__FIRAPP_DEFAULT: options, api_key=X, app_id=X:android:X, database_url=X, messaging_sender_id=X, storage_bucket=X, project_id=X
Added app name=__FIRAPP_DEFAULT: options, api_key=X, app_id=X, database_url=X, messaging_sender_id=X, storage_bucket=X, project_id=X
(Filename: Assets/Scripts/Firebase/FirebaseInitializer.cs Line: 29)

Did not register exception handlers: Crashlytics SDK was not initialized
UnityEngine.StackTraceUtility:ExtractStackTrace ()
UnityEngine.DebugLogHandler:LogFormat (UnityEngine.LogType,UnityEngine.Object,string,object[])
UnityEngine.Logger:Log (UnityEngine.LogType,object)
UnityEngine.Debug:Log (object)
Firebase.Platform.FirebaseLogger:LogMessage (Firebase.Platform.PlatformLogLevel,string) (at Z:/tmp/tmp.pKCgWTopY4/firebase/app/client/unity/src/Unity/FirebaseLogger.cs:89)
Firebase.LogUtil:LogMessage (Firebase.LogLevel,string) (at Z:/tmp/tmp.teccva8Fjv/firebase/app/client/unity/proxy/LogUtil.cs:68)
Firebase.Crashlytics.Crashlytics/PlatformAccessor:.cctor () (at Z:/tmp/tmp.U7RyIC0cPf/firebase/crashlytics/client/unity/proxy/Crashlytics.cs:123)
Firebase.Crashlytics.Crashlytics:Initialize () (at Z:/tmp/tmp.U7RyIC0cPf/firebase/crashlytics/client/unity/proxy/Crashlytics.cs:59)
System.Reflection.MonoMethod:Invoke (object,System.Reflection.BindingFlags,System.Reflection.Binder,object[],System.Globalization.CultureInfo)
System.Reflection.MethodBase:Invoke (object,object[])
Firebase.FirebaseApp:InitializeCrashlyticsIfPresent () (at Z:/tmp/tmp.teccva8Fjv/firebase/app/client/unity/proxy/FirebaseApp.cs:526)
Firebase.FirebaseApp:<CreateAndTrack>m__1 () (at Z:/tmp/tmp.teccva8Fjv/firebase/app/client/unity/proxy/FirebaseApp.cs:654)
Firebase.Dispatcher:Run<bool> (System.Func`1<bool>) (at Z:/tmp/tmp.pKCgWTopY4/firebase/app/client/unity/src/Platform/Dispatcher.cs:40)
Firebase.Platform.FirebaseHandler:RunOnMainThread<bool> (System.Func`1<bool>) (at Z:/tmp/tmp.pKCgWTopY4/firebase/app/client/unity/src/Unity/FirebaseHandler.cs:115)
Firebase.FirebaseApp:CreateAndTrack (Firebase.FirebaseApp/CreateDelegate,Firebase.FirebaseApp) (at Z:/tmp/tmp.teccva8Fjv/firebase/app/client/unity/proxy/FirebaseApp.cs:652)
Firebase.FirebaseApp:Create () (at Z:/tmp/tmp.teccva8Fjv/firebase/app/client/unity/proxy/FirebaseApp.cs:124)
Firebase.FirebaseApp:get_DefaultInstance () (at Z:/tmp/tmp.teccva8Fjv/firebase/app/client/unity/proxy/FirebaseApp.cs:99)
FirebaseInitializer:<Start>b__3_0 (System.Threading.Tasks.Task`1<Firebase.DependencyStatus>) (at Assets/Scripts/Firebase/FirebaseInitializer.cs:29)
Firebase.Extensions.TaskExtension/<ContinueWithOnMainThread>c__AnonStorey8`1/<ContinueWithOnMainThread>c__AnonStorey9<Firebase.DependencyStatus>:<>m__0 () (at Z:/tmp/tmp.7JEOGUoEwk/firebase/app/client/unity/task-extension/TaskExtension.cs:128)
Firebase.Dispatcher/<RunAsync>c__AnonStorey1`1<bool>:<>m__0 () (at Z:/tmp/tmp.pKCgWTopY4/firebase/app/client/unity/src/Platform/Dispatcher.cs:77)
Firebase.ExceptionAggregator:Wrap (System.Action) (at Z:/tmp/tmp.pKCgWTopY4/firebase/app/client/unity/src/Platform/ExceptionAggregator.cs:112)
Firebase.Dispatcher:PollJobs () (at Z:/tmp/tmp.pKCgWTopY4/firebase/app/client/unity/src/Platform/Dispatcher.cs:123)
Firebase.Platform.FirebaseHandler:Update () (at Z:/tmp/tmp.pKCgWTopY4/firebase/app/client/unity/src/Unity/FirebaseHandler.cs:207)
Firebase.Platform.FirebaseMonoBehaviour:Update () (at Z:/tmp/tmp.pKCgWTopY4/firebase/app/client/unity/src/Unity/FirebaseMonoBehaviour.cs:45)

(Filename: Assets/Scripts/Firebase/FirebaseInitializer.cs Line: 29)

DEBUG: Creating Auth  for App 
DEBUG: Auth state changed. Notifying 3 listeners.
DEBUG: ID token changed. Notifying 2 listeners.
Disabling all app initializers

Disable analytics

Disable auth

Disable database

Disable dynamic_links

Disable functions

Disable instance_id

Disable messaging

Disable remote_config

Disable storage

Auth state changed. Notifying 3 listeners.

Signed in XXXXX

Would set user identifier if running on a physical device: XXXXXX

ID token changed. Notifying 2 listeners.

[MODES] ModeService[default].RefreshMenus
[MODES] ModeService[default].UpdateModeMenus
Unloading 351 Unused Serialized files (Serialized files now loaded: 0)
UnloadTime: 1.980800 ms
Loaded scene 'Assets/Scenes/Menu.unity'
```

#### Relevant Code:
My current firebase init init:
```
void Start()
    {
        Firebase.FirebaseApp.LogLevel = Firebase.LogLevel.Debug;
        Firebase.FirebaseApp.CheckAndFixDependenciesAsync().ContinueWithOnMainThread(task =>
        {
            try
            {
                var firebaseApp = FirebaseApp.DefaultInstance;  // THIS IS LINE 29
                dependencyStatus = task.Result;
                if (dependencyStatus == Firebase.DependencyStatus.Available)
                {
                    authManager.InitializeFirebase();
                    troopsState.pullTroopsData();
                }
                else
                {
                    Debug.LogError(
                      "Could not resolve all Firebase dependencies: " + dependencyStatus);
                }
            }
            catch (System.Exception e)
            {
                Debug.LogException(e, this);
                throw;
            }
        });
   }
```

**Comment by paulinon** [comment]
Hi @demianardus,

In order for you to test if Crashlytics is working, you have to build your app on your phone. Remember that Crashlytics reports will be sent after your app is relaunched.

Additionally, instead of waiting for your app to crash on its own, you can try to [force a non-fatal exception](https://firebase.google.com/docs/crashlytics/test-implementation?platform=unity#force_a_crash_to_test_your_implementation) so that you can ensure that Crashlytics is working.

I'll be closing this for now. Feel free to reopen if you have any questions or if the issue persists.

Thanks!

**Comment by demianardus** [comment]
@paulinon thanks for your quick answer. I wasn't sure that was the case. 

I built this version and deployed it to my android internal test with a forced exception (`throw new System.Exception("test exception")`), and I did not get the report on the Dashboard (the app did get stuck).

I then started thinking that probably if Crashlytics was saying that it was not initializing properly on Unity Editor, the same thing was potentially going on on my Android device.
